iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -166,22 +172,28 @@ inline void swap(storable_function<Capacity, T>& f, storable_function<Capacity,
     f.swap(g);
 }
 
+template <typename T>
+void* allocate(byte_t* startAddress, uint64_t Capacity)
+{
+    // NOLINTBEGIN(cppcoreguidelines-pro-type-reinterpret-cast, performance-no-int-to-ptr) required for low level pointer alignment
+    uint64_t alignment = alignof(T);
+    uint64_t alignedPosition = cxx::align(reinterpret_cast<uint64_t>(startAddress), alignment);
+    cxx::Expects(alignedPosition + sizeof(T) - reinterpret_cast<uint64_t>(startAddress) < Capacity
```

Comment:
@elBoberido @FerdinandSpitzschnueffler

Now we have a regression. Previously it could be detected at compile time that the allocation will fail due to insufficient size for a callable (if used with some type that was too large). This was one reason for the static storage. Now we have checks at runtime for this, which leads to some issues which are just not necessary.

1.) Performance overhead (if the check is enabled, which it currently is)
2.) We may not know at compile time that we have insufficient space and cannot guarantee that it runs at compile time (with sufficient stack, but this is still required).

This is not good, we want to follow a fail early philosophy and you cannot really fail much earlier than compile time. Note that we could add the compile time check here also directly, as it was done in `static_storage`. However, I do not quite see the improvement. We shifted more of the storage responsibility to `function`, instead of abstracting it in something like `static_storage`.

Reply:
I think we can improve this function so that it is compile time again.
1. `Capacity` becomes the second template argument
2. We add this `static_assert(sizeof(T) + alignof(T) - 1 <= Capacity, "type does not fit into storage");`

With this it should be possible to remove the `cxx::Expects`.

Reply:
@MatthiasKillat yes, we replace the dependency on one class with another class but in total there is less code since one dependency had a large overlap with the one that replaced it.

Reply:
@MatthiasKillat @elfenpiff I added the compile time check.